### ISSUE TITLE
docker/connection.rb doesn't handle client errors

### ISSUE THREAD

**Issue by wyaeld** [swipely/docker-api#273]
I'm getting an error using Docker::Image remove method

You can see the example below, but the cause appears to be https://github.com/swipely/docker-api/blob/master/lib/docker/connection.rb#L84 which is not included the 400+ range HTTP status which docker can return.  Conflict being a 409

In this case, I know I'm trying to remove an image layer which has a container, so the server doesn't let me, so the ConflictError is desired behavior

```bash
$ Removing dd3b7896c13527a9039962c95454f9bee008d5ef25a72f08b3397fe50c0bdd61 ["dockerfile/ghost:latest"] because ["ghost"]

/home/wyaeld/.rvm/gems/ruby-2.2.1/gems/docker-api-1.21.0/lib/docker/connection.rb:52:in `rescue in request': Expected([200, 201, 202, 203, 204, 304]) <=> Actual(409 Conflict) (Docker::Error::ConflictError)
	from /home/wyaeld/.rvm/gems/ruby-2.2.1/gems/docker-api-1.21.0/lib/docker/connection.rb:38:in `request'
	from /home/wyaeld/.rvm/gems/ruby-2.2.1/gems/docker-api-1.21.0/lib/docker/connection.rb:61:in `block (2 levels) in <class:Connection>'
	from /home/wyaeld/.rvm/gems/ruby-2.2.1/gems/docker-api-1.21.0/lib/docker/image.rb:69:in `remove'

```

From https://github.com/docker/docker/blob/c2f82bba0a7d365ae4712804d9aed50dff788898/docs/sources/reference/api/docker_remote_api_v1.4.md

Seems you can have the following client errors:

400 – bad parameter
404 – no such container / image
406 – impossible to attach (container not running)
409 – conflict between containers and images

**Comment by tlunter** [comment]
Have you rescued with `Docker::Error::ConflictError`? You shouldn't be expecting a 409 from a successful HTTP request. It looks to me that we're appropriately raising the ConflictError.

Maybe I don't understand, what is the preferred behavior?

**Comment by wyaeld** [comment]
The error is always this though

#<Docker::Error::ConflictError: Expected([200, 201, 202, 203, 204, 304]) <=> Actual(409 Conflict)

That appears to be an issue with the contract enforced by docker-api, since thats where the expects is defined.  Shouldn't the conflict error contain info about the conflict?

**Comment by tlunter** [comment]
@wyaeld I released an update in v1.21.1 that should make the errors much more descriptive!

**Comment by wyaeld** [comment]
Excellent.  I can confirm that's working

```
#<Docker::Error::ConflictError: Conflict, cannot delete dd3b7896c135 because the container e32197ed552e is using it, use -f to force
```

Now I'm able to parse out the ids of the conflicting containers.

Thanks for the help.
